lib/Mojolicious/Plugin/DefaultHelpers.pm

```diff
@@ -161,6 +165,61 @@ sub _is_fresh {
   return $c->app->static->is_fresh($c, \%options);
 }
 
+sub _proxy_method_p {
+  my ($method, $c) = (shift, shift);
+  return _proxy_start_p($c, $c->ua->build_tx($method, @_));
+}
+
+sub _proxy_start_p {
+  my ($c, $source_tx, $cb) = @_;
+  my $tx = $c->render_later->tx;
+
+  my $promise = Mojo::Promise->new;
+  $source_tx->res->content->auto_upgrade(0)->auto_decompress(0)->once(
+    body => sub {
+      my $source_content = shift;
+
+      my $source_res = $source_tx->res;
+      my $res        = $tx->res;
+      my $content    = $res->content;
+      $res->code($source_res->code)->message($source_res->message);
+      my $headers = $source_res->headers->clone->dehop;
+      $content->headers($headers);
+      $c->$cb($source_tx) if $cb;
```

Comment:
I feel it's a bit awkward to pass in a callback. Could we (ab)use events on the transaction instead? Like `$source_tx->emit('proxy_start')` or something?

Reply:
That event would be specific to a Mojolicious level experimental helper? How do you describe that event in the documentation?

Reply:
I thought about the same (issue about documentation), but I still think adding the specific documentation together with the helper is less strange than the callback. I also wonder if adding more hooks/events in the future could make more sense.

Reply:
Yes i agree, if we can find a good justification for the event it would be much less awkward. But we have to find a justification that's not a high level helper.

Reply:
The same way making promises actually work for this would be less awkward. But we can't, so here we are with the callback being the best choice at the moment.

Reply:
I'm glad you managed to get rid of $cb 👍